src/main/kotlin/org/malachite/estella/queues/utils/MsgDeserializer.kt

```diff
@@ -0,0 +1,43 @@
+package org.malachite.estella.queues.utils
+
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import org.malachite.estella.commons.models.tasks.TaskResult
+import org.malachite.estella.process.domain.toTask
+import org.malachite.estella.services.TaskService
+import org.malachite.estella.services.TaskStageService
+import org.springframework.beans.factory.annotation.Autowired
+import java.sql.Timestamp
+import java.util.*
+import javax.sql.rowset.serial.SerialBlob
+import javax.sql.rowset.serial.SerialClob
+
+class MsgDeserializer(
+        @Autowired private val taskService: TaskService,
+        @Autowired private val taskStageService: TaskStageService) {
+
+        fun toTaskResult(msg: String): TaskResult? {
+            try {
+                val based = Base64.getDecoder().decode(msg).decodeToString()
+                val decodedMsg = Json.decodeFromString<Map<String, String>>(based)
```

Comment:
Zrób sobie jakąś klasę pomocniczą, w stylu `RabbitTaskDto`, która będzie posiadała wszystkie pola, które dostaniesz z rabbita. Wtedy zamiast się pierdzielić z mapami, będziesz mógł zrobić coś takiego:
```kotlin
Json.decodeFromString<RabbitTaskDto>(based)
```
Ten sposób od razu też Ci rzuci wyjątek, jeśli nie ma jakiejś wartości. Plus jeśli używasz tego based tylko tutaj, polecam użyć leta
```kotlin
val decoded = Base64.getDecoder().decode(msg).decodeToString().let{
   Json.decodeFromString<RabbitTaskDto>(it)
}
```
Wtedy zwrócisz sobie tylko decoded, a resztę wykonasz w consumerze, żeby zachować czystość rasową tej klasy. Będziemy wtedy też mieli to rozróżnienie, o którym mówiłem wczoraj - tu będziesz miał wyjątek deserializacji, w konsumerze będziesz miał wyjątek o tym, że jest czegoś brak.

Reply:
czy wobec tego na pewno warto rozbijać to na dwie klasy, jeśli ta będzie miała tylko coś typu:
```kotlin
    fun toTaskResult(msg: String): TaskResult? {
        try {
            return Base64.getDecoder().decode(msg).decodeToString().let { Json.decodeFromString<TaskResultRabbitDTO>(it) }
        } catch (ex: Exception) {
            ex.printStackTrace()
            println("DBG: Bad request. Couldn't parse msg to TaskResult: $msg")
            return null
        }
    }
```
?